src/main/java/pl/eukon05/eventboard/event/application/service/InviteToEventUseCase.java

```diff
@@ -0,0 +1,43 @@
+package pl.eukon05.eventboard.event.application.service;
+
+import lombok.RequiredArgsConstructor;
+import pl.eukon05.eventboard.common.UseCase;
+import pl.eukon05.eventboard.event.application.port.in.GetUserFriendlistPort;
+import pl.eukon05.eventboard.event.application.port.out.GetEventPort;
+import pl.eukon05.eventboard.event.application.port.out.SaveEventPort;
+import pl.eukon05.eventboard.event.domain.Event;
+import pl.eukon05.eventboard.event.domain.EventType;
+
+import java.util.List;
+import java.util.Optional;
+
+@UseCase
+@RequiredArgsConstructor
+class InviteToEventUseCase {
+    private final GetUserFriendlistPort getUserFriendlistPort;
+    private final GetEventPort getEventPort;
+    private final SaveEventPort saveEventPort;
+
+    boolean execute(String selfID, String friendID, long eventID) {
+        Optional<Event> eventOptional = getEventPort.getEventById(eventID);
+
+        if (eventOptional.isEmpty()) return false;
+
+        Event event = eventOptional.get();
+
+        List<String> friendlist = getUserFriendlistPort.getUserFriendlist(selfID);
```

Comment:
If the port is only used to check if someone is friend of someone why do why have to return list of all friends?

```suggestion
        friendsPort.areUsersFriends(selfID, friendId);
```

Reply:
Now that you mention it, I don't know either 😅
I'll re-do it to return a boolean value, depending on if the user in question is a "friend" or not